phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java

```diff
@@ -655,13 +678,20 @@ public void close() throws SQLException {
                     traceScope.close();
                 }
                 closeStatements();
+                if(childConnections != null) {
+                    SQLCloseables.closeAllQuietly(childConnections);
```

Comment:
This is guaranteed to only ever happen from 1 thread at a time I suppose? Which is why we wouldn't care about thread-safety of `closeAllQuietly`?

Reply:
But the check above it should also be synchronized i.e. `if(childConnections != null) {` no?

Reply:
Yep agreed, did this too late at night.  I've synchronized the creation/creation checks of this structure.